core/src/test/scala/unit/kafka/server/ReplicaFetcherTierStateMachineTest.scala

```diff
@@ -121,4 +184,106 @@ class ReplicaFetcherTierStateMachineTest {
     assertTrue(fetcher.fetchState(partition).isEmpty)
     assertTrue(failedPartitions.contains(partition))
   }
+
+  private def mockBuildRemoteLogAuxState(mockReplicaMgr: ReplicaManager, topicPartition: TopicPartition): Unit = {
+    val idPartition = new TopicIdPartition(Uuid.randomUuid(), topicPartition)
+    logDir = JTestUtils.tempDirectory(s"kafka-${this.getClass.getSimpleName}")
+    val tpDir = JTestUtils.tempDirectory(logDir.toPath, idPartition.toString)
+    stateManager = new ProducerStateManager(topicPartition, tpDir, 5 * 60 * 1000, producerStateManagerConfig, time)
+    val unifiedLog = buildUnifiedLog(topicPartition, stateManager, tpDir)
+
+    doReturn(unifiedLog, Nil: _*).when(mockReplicaMgr).localLogOrException(any(classOf[TopicPartition]))
```

Comment:
For my understanding, what is the purpose of `Nil: _*`? Is it a roundabout way to verify that the method is called only once because on the second call you would get a NullPointerException? If so, can you have an explicit verification of the number of invocations?